packages/mdc-icon-button/foundation.js

```diff
@@ -0,0 +1,179 @@
+      setText: (/* text: string */) => {},
+      getTabIndex: () => /* number */ 0,
+      setTabIndex: (/* tabIndex: number */) => {},
+      getAttr: (/* name: string */) => /* string */ '',
+      setAttr: (/* name: string, value: string */) => {},
+      removeAttr: (/* name: string */) => {},
+      notifyChange: (/* evtData: IconButtonToggleEvent */) => {},
+    };
+  }
+
+  constructor(adapter) {
+    super(Object.assign(MDCIconButtonToggleFoundation.defaultAdapter, adapter));
+
+    /** @private {boolean} */
+    this.on_ = false;
+
+    /** @private {boolean} */
+    this.disabled_ = false;
+
+    /** @private {number} */
+    this.savedTabIndex_ = -1;
+
+    /** @private {?IconButtonToggleState} */
+    this.toggleOnData_ = null;
+
+    /** @private {?IconButtonToggleState} */
+    this.toggleOffData_ = null;
+
+    this.clickHandler_ = /** @private {!EventListener} */ (
+      () => this.toggleFromEvt_());
+  }
+
+  init() {
+    this.refreshToggleData();
+    this.savedTabIndex_ = this.adapter_.getTabIndex();
+    this.adapter_.registerInteractionHandler('click', this.clickHandler_);
+  }
+
+  destroy() {
+    this.adapter_.deregisterInteractionHandler('click', this.clickHandler_);
+  }
+
+  refreshToggleData() {
+    this.toggleOnData_ = {
+      label: this.adapter_.getAttr(strings.DATA_TOGGLE_ON_LABEL),
+      content: this.adapter_.getAttr(strings.DATA_TOGGLE_ON_CONTENT),
+      cssClass: this.adapter_.getAttr(strings.DATA_TOGGLE_ON_CLASS),
+    };
+    this.toggleOffData_ = {
+      label: this.adapter_.getAttr(strings.DATA_TOGGLE_OFF_LABEL),
+      content: this.adapter_.getAttr(strings.DATA_TOGGLE_OFF_CONTENT),
+      cssClass: this.adapter_.getAttr(strings.DATA_TOGGLE_OFF_CLASS),
+    };
+  }
+
+  /** @private */
+  toggleFromEvt_() {
+    this.toggle();
+    const {on_: isOn} = this;
+    this.adapter_.notifyChange(/** @type {!IconButtonToggleEvent} */ ({isOn}));
+  }
+
+  /** @return {boolean} */
+  isOn() {
+    return this.on_;
+  }
+
+  /** @param {boolean=} isOn */
+  toggle(isOn = !this.on_) {
+    this.on_ = isOn;
+
+    const {ARIA_LABEL, ARIA_PRESSED} = MDCIconButtonToggleFoundation.strings;
+
+    this.adapter_.setAttr(ARIA_PRESSED, this.on_.toString());
+
+    const {cssClass: classToRemove} =
+        this.on_ ? this.toggleOffData_ : this.toggleOnData_;
+
+    if (classToRemove) {
+      this.adapter_.removeClass(classToRemove);
+    }
+
+    const {content, label, cssClass} = this.on_ ? this.toggleOnData_ : this.toggleOffData_;
+
+    if (cssClass) {
+      this.adapter_.addClass(cssClass);
+    }
+    if (content) {
+      this.adapter_.setText(content);
+    }
+    if (label) {
+      this.adapter_.setAttr(ARIA_LABEL, label);
+    }
+  }
+
+  /** @return {boolean} */
+  isDisabled() {
+    return this.disabled_;
+  }
+
+  /** @param {boolean} isDisabled */
+  setDisabled(isDisabled) {
+    this.disabled_ = isDisabled;
+
+    const {ARIA_DISABLED} = MDCIconButtonToggleFoundation.strings;
```

Comment:
We don't need to maintain aria-disabled anymore since we're stating that disabled state applies only to buttons directly, right? (Also affects tests, and probably renders this constant unused)